base/util.jl

```diff
@@ -278,11 +278,33 @@ end
 getpass(prompt::AbstractString) = getpass(stdin, stdout, prompt)
 
 """
-    prompt(message; default="") -> Union{String, Nothing}
+    prompt(message; default="", timeout=nothing) -> Union{String, Nothing}
 
 Displays the `message` then waits for user input. Input is terminated when a newline (\\n)
 is encountered or EOF (^D) character is entered on a blank line. If a `default` is provided
-then the user can enter just a newline character to select the `default`.
+then the user can enter just a newline character to select the `default`. A `timeout` in seconds
+greater than 0 can be provided, after which the default will be returned.
+
+For instance, the user enters a value and hits `return`:
+```julia
+julia> Base.prompt("Proceed? y/n"; default="n", timeout=5)
+Proceed? y/n [n] timeout 5 seconds: y
```

Comment:
I think the "timeout 5 seconds" part looks a bit strange or misplaced. Does it have to be displayed to the user at all?

Reply:
Seems useful to have but probably a clearer wording like "Proceed? y/n. Default value [n] will be used in 5 seconds: y" would be better?